serde-generate/tests/csharp_runtime.rs

```diff
@@ -0,0 +1,308 @@
+    let (dir, _tmp) = create_test_dir("test_csharp_bincode_runtime_on_simple_data");
+    test_csharp_runtime_on_simple_data(dir, Runtime::Bincode);
+}
+
+fn test_csharp_runtime_on_simple_data(dir: PathBuf, runtime: Runtime) {
+    let test_dir = copy_test_project(&dir, false);
+
+    let registry = test_utils::get_simple_registry().unwrap();
+
+    let installer = csharp::Installer::new(dir.clone());
+    installer.install_serde_runtime().unwrap();
+    installer.install_bincode_runtime().unwrap();
+    installer.install_lcs_runtime().unwrap();
+
+    // Generates code into `Serde/Tests`
+    let config =
+        CodeGeneratorConfig::new("Serde.Tests".to_string()).with_encodings(vec![runtime.into()]);
+    let generator = csharp::CodeGenerator::new(&config);
+    generator.write_source_files(dir, &registry).unwrap();
+
+    let reference = runtime.serialize(&Test {
+        a: vec![4, 6],
+        b: (-3, 5),
+        c: Choice::C { x: 7 },
+    });
+
+    let mut source = File::create(&test_dir.join("TestRuntime.cs")).unwrap();
+    writeln!(
+        source,
+        r#"
+using System;
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+
+namespace Serde.Tests {{
+    public class TestRuntime {{
+        [Test]
+        public void TestRoundTrip() {{
+            byte[] input = new byte[] {{{0}}};
+
+            Test test = Test.{1}Deserialize(input);
+
+            List<uint> a = new List<uint>(new uint[] {{ 4, 6 }});
+            var b = ((long)-3, (ulong)5);
+            Choice c = new Choice.C((byte) 7);
+            Test test2 = new Test(a, b, c);
+
+            Assert.AreEqual(test, test2);
+
+            byte[] output = test2.{1}Serialize();
+
+            CollectionAssert.AreEqual(input, output);
+
+            byte[] input2 = new byte[] {{{0}, 1}};
+            Assert.Throws<DeserializationException>(() => Test.{1}Deserialize(input2));
+        }}
+    }}
+}}
+"#,
+        reference
+            .iter()
+            .map(|x| format!("{}", *x as u8))
+            .collect::<Vec<_>>()
+            .join(", "),
+        runtime.name().to_camel_case(),
+    )
+    .unwrap();
+
+    dotnet_build(&test_dir);
+    run_nunit(&test_dir);
+}
+
+#[test]
+fn test_csharp_lcs_runtime_on_supported_types() {
+    let (dir, _tmp) = create_test_dir("test_csharp_lcs_runtime_on_supported_types");
+    test_csharp_runtime_on_supported_types(dir, Runtime::Lcs);
+}
+
+#[test]
+fn test_csharp_bincode_runtime_on_supported_types() {
+    let (dir, _tmp) = create_test_dir("test_csharp_bincode_runtime_on_supported_types");
+    test_csharp_runtime_on_supported_types(dir, Runtime::Bincode);
+}
+
+fn quote_bytes(bytes: &[u8]) -> String {
+    format!(
+        "{{{}}}",
+        bytes
+            .iter()
+            .map(|x| format!("{}", *x as u8))
+            .collect::<Vec<_>>()
+            .join(", ")
+    )
+}
+
+fn test_csharp_runtime_on_supported_types(dir: PathBuf, runtime: Runtime) {
```

Comment:
And now the extended runtime test with fuzzing for both LCS and bincode! Very nice!